src/appleseed/foundation/math/dual.h

```diff
@@ -66,6 +71,8 @@ class Dual
     const T& get_dx() const;
     const T& get_dy() const;
 
+    void set_derivatives(const T& dx, const T& dy);
```

Comment:
Don't we need a `set_value()` has well then?

Reply:
I added this method to avoid copying m_value when we compute the ray differentials.
I don't think we need set_value(), at least for now.

Reply:
Ok